sys/sys/pcpu.h

```diff
@@ -121,8 +124,9 @@ extern uintptr_t dpcpu_off[];
  * subtracted the DPCPU_START, so we do it now.
  */
 #define	_DPCPU_PTR(b, n)					\
-    (__typeof(DPCPU_NAME(n))*)((b) +                            \
-        ((vaddr_t)&DPCPU_NAME(n) - (vaddr_t)DPCPU_START))
+	cheri_setboundsexact((__typeof(DPCPU_NAME(n)) *)((b) +	\
```

Comment:
Could we also end up triggering https://github.com/CTSRD-CHERI/llvm-project/issues/495 here?

Reply:
Yes, but I think all the current variables are small enough, so I'll just leave it as-is for now.  I'm hoping we can fix that issue before we get any dpcpu variables large enough to trigger this.